src/components/Tree/child.js

```diff
@@ -19,26 +19,56 @@ export default function Child(props) {
         isChecked,
         icon,
         nodePath,
-        onExpandCollapse,
-        onSelect,
+        onNodeExpand,
+        onNodeCheck,
+        onNodeSelect,
+        isSelected,
+        name,
+        selectedNode,
     } = props;
     const hasChildren = Array.isArray(children);
     const hasCheckbox = typeof isChecked === 'boolean' || isChecked === 'indeterminate';
     const hasIcon = !!icon;
+    const levelMatch = name.match(/\./g);
+    const ariaLevelValue = name && levelMatch ? levelMatch.length + 1 : 1;
+    const tabIndex = isSelected ? 0 : -1;
+
+    const handleNodeSelect = event => {
+        event.stopPropagation();
+        onNodeSelect({ name });
+    };
+
+    const handleExpandCollapse = () => {
+        return onNodeExpand({ nodePath });
+    };
+
     return (
-        <ItemContainerLi hasChildren={hasChildren} icon={icon} data-id="node-element-li">
-            <NodeContainer data-id="node-element">
+        <ItemContainerLi
+            id={name}
+            data-id="node-element-li"
+            icon={icon}
+            hasChildren={hasChildren}
+            onClick={handleNodeSelect}
+            role="treeitem"
+            aria-label={hasChildren ? label : undefined}
```

Comment:
when discussed we said that this will not be used now, also label can be a node and you can't pass a react node as an attribute to an element